nemo_skills/inference/model/code_execution.py

```diff
@@ -208,27 +218,21 @@ def _generate_single(
             # .rfind(code_end, 0, -1) searches for the second-to-last occurrence of code_end and checks
             # that the last code_begin is not closed to ensure that we are inside the code block
             if output.endswith(code_end) and output.rfind(code_begin) > output.rfind(code_end, 0, -1):
-                code_execution_time_start = time.time()
-                execution_dict, session_id = self.sandbox.execute_code(
-                    generated_code=extract_code_to_execute(output, code_begin, code_end),
-                    timeout=self.config.code_execution_timeout,
-                    max_output_characters=self.config.max_code_output_characters,
-                    session_id=session_id,
-                    traceback_verbosity=self.config.sandbox_traceback_verbosity,
-                )
+                code_execution_time_start, execution_dict = self.execute_generated_code(prompt, code_begin, code_end, output, session_id)
                 remaining_code_executions = None
                 if self.config.add_remaining_code_executions:
                     remaining_code_executions = effective_max_code_executions - generation_index - 1
                 # adding code output to the prompt
                 code_output = format_code_output(
                     execution_dict, code_output_begin, code_output_end, code_output_format, remaining_code_executions
                 )
-                
+                request['tokens_to_generate'] = max_new_tokens - len(self.tokenizer.encode(request['prompt'], add_special_tokens=True))
```

Comment:
I don't think this is the right change. tokens_to_generate means whatever comes after prompt. So we shouldn't subtract the prompt, but should subtract the code output. But in any case, let's probably remove it from this pr

Reply:
The issue I was having, and the reason I put this here, is that I was regularly getting errors that break the whole generation loop when the entire context length exceeds the model max (32k tokens I believe, in the case of LIMO). So this was to prevent issues when, due to output being formatted in, the "tokens to generate" plus existing content would  exceed model max and break an entire generation run.

That being said, this can probably also be reasonably controlled with overall max tokens to generate. I'll remove it for now and run another test with LIMO to make sure.

Reply:
Okay I updated and it seems fine now with this code removed--it's possible it had interactions with other bugs in my initial implementation. I am running a longer run just to make sure.

Reply:
@Kipok Okay longer run has been running for a while and it seems everything is effectively controlled with `++inference.tokens_to_generate`